src/lib/broadcast.jl

```diff
@@ -272,7 +272,8 @@ end
 
   end
 
-  @adjoint (::Type{T})(xs::Array) where {T <: CUDA.CuArray} =
+  _eyelike(y::CUDA.CuVector{T}) where T = CUDA.CuArray(I(length(y)))
```

Comment:
```suggestion
  _eyelike(y::CUDA.CuVector) = CUDA.CuArray(I(length(y)))
```
To avoid the unbound type param.